src/pages/index.js

```diff
@@ -15,16 +15,83 @@ import {
   cardList,
   templateSelector,
   options,
+  profileTitle,
+  profileImage,
+  profileDescription,
+  updateAvatarContainer,
+  updateAvatarForm,
 } from "../utils/constants.js";
 import "./index.css";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
+//////////////////////
+// ----- APIs ----- //
+//////////////////////
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  headers: {
+    authorization: "02a3c077-82c6-4270-8df6-e8c1f2fec9e8",
+    "content-type": "application/json",
+  },
+});
+
+api
+  .getUserInformation()
+  .then((data) => {
+    profileTitle.textContent = data.name;
+    profileDescription.textContent = data.about;
+    profileImage.src = data.avatar;
+  })
+  .catch((err) => console.error(err));
+
+api
+  .getInitialCards()
+  .then((data) => {
+    cardSection.renderItems(data.reverse());
+  })
+  .catch((err) => console.error(err));
+
+//////////////////////////
 // ----- SECTIONS ----- //
+//////////////////////////
 
 function createCard(item) {
-  const cardElement = new Card(item, templateSelector, (data) => {
-    previewPicturePopup.open(data);
-  });
-  return cardElement.getView();
+  const card = new Card(
+    item,
+    templateSelector,
+    (data) => {
+      previewPicturePopup.open(data);
+    },
+    (data) => {
+      deleteCardPopup.open();
+      // set the submit action
+      deleteCardPopup.setSubmitAction(() => {
+        deleteCardPopup.renderLoadingMessage("Deleting...");
+        api
+          .handleDeleteCard(data._cardID)
+          .then(() => {
+            card.handleDeleteCard();
+          })
+          .then(() => {
+            deleteCardPopup.close();
+            deleteCardPopup.resetSaveButton("Yes");
```

Comment:
Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally